martin-mbtiles/src/mbtiles.rs

```diff
@@ -378,32 +388,102 @@ impl Mbtiles {
         Err(MbtError::NoUniquenessConstraint(self.filepath.clone()))
     }
 
-    pub async fn validate_mbtiles<T>(&self, conn: &mut T) -> MbtResult<()>
+    pub async fn generate_global_hash<T>(&self, conn: &mut T) -> MbtResult<()>
     where
         for<'e> &'e mut T: SqliteExecutor<'e>,
     {
         let mbttype = self.detect_type(&mut *conn).await?;
 
-        let sql = match mbttype {
-            MbtType::Flat => {
-                return Err(IncorrectDataFormat(
-                    self.filepath().to_string(),
-                    &[MbtType::FlatWithHash, MbtType::Normalized],
-                    MbtType::Flat,
-                ));
-            }
-            MbtType::FlatWithHash => {
-                "SELECT * FROM tiles_with_hash WHERE tile_hash!=hex(md5(tile_data)) LIMIT 1;"
+        let select_from = match mbttype {
+            Flat => {
+                println!("Cannot generate global hash, no hash column in flat table format. Skipping global_hash generation...");
+                return Ok(());
             }
-            MbtType::Normalized => {
-                "SELECT * FROM images WHERE tile_id!=hex(md5(tile_data)) LIMIT 1;"
+            MbtType::FlatWithHash => "SELECT 'global_hash', hex(md5_concat(tile_hash)) FROM tiles_with_hash ORDER BY zoom_level, tile_column, tile_row;",
+            MbtType::Normalized => "SELECT 'global_hash', hex(md5_concat(images.tile_id)) FROM images JOIN map ON images.tile_id=map.tile_id ORDER BY zoom_level, tile_column, tile_row;"
+
+        }.to_string();
+
+        let rusqlite_conn = RusqliteConnection::open(Path::new(&self.filepath()))?;
+        register_md5_function(&rusqlite_conn)?;
+        rusqlite_conn.execute(
+            format!("INSERT OR REPLACE INTO metadata(name, value) {select_from}").as_str(),
```

Comment:
i think it would be better to pass `global_hash` as a parameter here, and replace it above with a `?`. This way it is only declared once. Besides, I am still thinking of a better name... Perhaps we should name it `md5sum` or `all_tiles_md5` or ... -- point being that we may want to mention the hashing type in its name

Reply:
I think for now since there is only one hash function, and the column in `FlatWithHash` is plainly called `tile_hash`, it should just stay `global_hash`

Reply:
Per additional discussion, it only makes sense to hash `z,x,y, (tile hash or tile blob or both)`. We will need to do perf test later, but for now lets do `z,x,y,hash`. Note that all integers will need to be converted to strings first - hashing is only available on strings and blobs.

Reply:
We can store which hash it is in the metadata table once we allow for different algorithms. Added a comment about this to the relevant 
issue: https://github.com/maplibre/martin/issues/835#issuecomment-1699677255